lib/scholar/naive_bayes/multinomial_naive_bayes.ex

```diff
@@ -0,0 +1,387 @@
+      case Nx.shape(sample_weights) do
+        {} ->
+          Nx.broadcast(sample_weights, {num_samples})
+
+        {^num_samples} ->
+          sample_weights
+
+        _ ->
+          raise ArgumentError,
+                "number of weights must match number of samples. Number of weights: #{Nx.size(sample_weights)} does not match number of samples: #{num_samples}"
+      end
+
+    classes_encoded = Nx.iota({num_classes})
+
+    classes =
+      y
+      |> Scholar.Preprocessing.ordinal_encode(num_classes: num_classes)
+      |> Scholar.Preprocessing.one_hot_encode(num_classes: num_classes)
+
+    {_, classes_features} = Nx.shape(classes)
+
+    classes =
+      cond do
+        classes_features == 1 and num_classes == 2 ->
+          Nx.concatenate([1 - classes, classes], axis: 1)
+
+        classes_features == 1 and num_classes != 2 ->
+          Nx.broadcast(1.0, Nx.shape(classes))
+
+        true ->
+          classes
+      end
+
+    classes =
+      if opts[:sample_weights_flag],
+        do: classes * Nx.reshape(sample_weights, {:auto, 1}),
+        else: classes
+
+    {_, n_classes} = Nx.shape(classes)
+    class_count = Nx.broadcast(0.0, {n_classes})
+    feature_count = Nx.broadcast(0.0, {n_classes, num_features})
+    feature_count = feature_count + Nx.dot(Nx.transpose(classes), x)
+    class_count = class_count + Nx.sum(classes, axes: [0])
+    alpha = check_alpha(Nx.tensor(opts[:alpha]), opts[:force_alpha], num_features)
+    smoothed_feature_count = feature_count + alpha
+    smoothed_cumulative_count = Nx.sum(smoothed_feature_count, axes: [1])
+
+    feature_log_probability =
+      Nx.log(smoothed_feature_count) - Nx.log(Nx.reshape(smoothed_cumulative_count, {:auto, 1}))
+
+    {class_priors_length} = Nx.shape(class_priors)
+
+    if num_classes != class_priors_length do
+      raise ArgumentError, "Number of priors must match number of classes."
+    end
+
+    class_log_priors =
+      cond do
+        opts[:priors_flag] ->
+          Nx.log(class_priors)
+
+        opts[:fit_priors] ->
+          Nx.log(class_count) - Nx.log(Nx.sum(class_count))
+
+        true ->
+          Nx.broadcast(-Nx.log(num_classes), {num_classes})
+      end
+
+    %__MODULE__{
+      classes: classes_encoded,
+      class_count: class_count,
+      feature_log_probability: feature_log_probability,
+      feature_count: feature_count,
+      class_log_priors: class_log_priors
+    }
+  end
+
+  defn joint_log_likelihood(
+         %__MODULE__{
+           feature_log_probability: feature_log_probability,
+           class_log_priors: class_log_priors
+         },
+         x
+       ) do
+    Nx.dot(x, [1], feature_log_probability, [1]) + class_log_priors
+  end
+
+  defnp check_alpha(alpha, force_alpha, num_features) do
+    alpha_lower_bound = 1.0e-10
+
+    case Nx.shape(alpha) do
+      {} -> nil
+      {^num_features} -> nil
+      _ -> raise ArgumentError, "When alpha is a list it should contain num_features values."
```

Comment:
Error messages usually start in lowercase and do not end with dot:

```suggestion
      _ -> raise ArgumentError, "when alpha is a list it should contain num_features values"
```

Reply:
Sorry, force of habit